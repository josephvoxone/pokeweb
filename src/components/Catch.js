import React, { useState } from 'react';
import styled from '@emotion/styled'
import { AddPokemon, getPokemon } from '../service/providers/Pokemon';
import { useHistory } from 'react-router-dom';

const Catch = (props) => {

    const pokemon = props.pokemon
    const history = useHistory();
    const [Valid, setValid] = useState()

    const Title = styled.h3`
        margin-top: 100px;
        margin-bottom: 5px;
        color: white;
    `;

    const SubTitle = styled.p`
        margin-top: 0;
        font-weight: 400;
        color: #f1f1f1;
    `;

    const Form = styled.form`
        display: inline-grid;
    `;

    const BtnPoke = styled.button`
        outline: none;
        border: none;
        padding: 10px 20px;
        border-radius: 7px;
        margin-top: 10px;
        background: linear-gradient(
    45deg
    , #e5c823, #ffbf00);
        font-size: 16px;
        font-weight: 700;
        cursor: pointer;
        transition: background 0.5s;
        letter-spacing: 1px;

        &:hover { 
            transition: background 0.5s;
            background: linear-gradient(
                45deg
                , #fff3af, #ffeb3b);
        }
    `;

    const NickInput = styled.input`
        padding: 10px 30px;
        outline: none;
        border: none;
        border-radius: 6px;
        font-weight: 500;
        font-size: 10pt;
    `;

    const Warn = styled.div`
        padding: 10px 30px;
        outline: none;
        border: none;
        border-radius: 6px;
        font-weight: 500;
        font-size: 10pt;
    `;



    const OnClickSubmit = (e) => {
        e.preventDefault()

        const newData = {
            id: props.pokemon.id,
            nickname: e.target.nickname.value.trim(),
            pokemon: props.pokemon.name,
        }

        const find = getPokemon.some((pokemon) => {
            return pokemon.nickname === e.target.nickname.value.trim()
        })

        setValid(!find)

        if (!find) {
            AddPokemon(newData);
            history.go(0)
        }

    }

    return (
        <>
            <Title>Congratulation, you got {pokemon.name}!</Title>
            <SubTitle>fill in the nickname to get this pokemon</SubTitle>
            <Form onSubmit={OnClickSubmit}>
                <NickInput autoComplete="off" type="text" name="nickname" placeholder="Nickname" maxLength="10" required></NickInput>
                {Valid === false ? <Warn>Oops, Please use different name!</Warn> : ''}
                <BtnPoke type="submit">SUBMIT</BtnPoke>
            </Form>
        </>
    )
}

export default Catch;