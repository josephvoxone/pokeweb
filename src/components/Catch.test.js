import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Catch from './Catch';

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("Test : Catch Pokemon", () => {
    {
        act(() => {
            const fakeData = {
                id: 1,
                name: 'Bulbasaur',
            }
            render(
                <Catch pokemon={fakeData}></Catch>
                , container)
        })
        expect(container).toMatchSnapshot()
    }
});
