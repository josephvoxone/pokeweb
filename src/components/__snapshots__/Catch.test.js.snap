// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test : Catch Pokemon 1`] = `
<div>
  <h3
    class="css-12oe26j"
  >
    Congratulation, you got 
    Bulbasaur
    !
  </h3>
  <p
    class="css-v0hj7n"
  >
    fill in the nickname to get this pokemon
  </p>
  <form
    class="css-5c4cj1"
  >
    <input
      autocomplete="off"
      class="css-1uyx8bt"
      maxlength="10"
      name="nickname"
      placeholder="Nickname"
      required=""
      type="text"
    />
    
    <button
      class="css-1q7abdy"
      type="submit"
    >
      SUBMIT
    </button>
  </form>
</div>
`;
